components/external_storage/src/s3.rs

```diff
@@ -27,22 +27,16 @@ use kvproto::backup::S3 as Config;
 pub struct S3Storage {
     config: Config,
     client: S3Client,
-    // The current implementation (rosoto 0.43.0 + hyper 0.13.3) is not `Send`
-    // in practical. See more https://github.com/tikv/tikv/issues/7236.
-    // FIXME: remove it.
-    _not_send: PhantomData<*const ()>,
```

Comment:
Without this phantom, `S3Storage` is error-prone, it may cause panic if it is used in other threads.

Reply:
From my reading the underlying github issues, the problem is from using the global dispatcher. The code below here is changed to instead create our own dispatcher. Let me know if you have any specific tests for this that I could run.

Reply:
@gregwebs I mainly have two questions for the issue, do you mind explain a bit?
1. how does `tokio::runtime::Builder::basic_scheduler()` work? From the document it says the runtime will execute tasks on "current thread". What if I create a basic runtime on thread T1, send it over to thread T2, and execute some task? Does T1 or T2 count for "current thread"?
2. how does creating a new dispatcher for each `AmsKms` and `S3Storage` instances (which are `Sync + Send`, so they allow people to send them to different threads)  fixes the issue?

Reply:
1) I don't know. The behavior after this PR is merged should be the same as previous.

2) My reading of the issue is that there is a global dispatcher that is initialized with lazy_static and it turns out it is not actually safe to use it. But instead you can just create a dispatcher at runtime yourself.

Reply:
This is the default shared client: https://github.com/rusoto/rusoto/blob/d59d716f097f64a713b3bbcd5fc7b84e210647e4/rusoto/core/src/client.rs#L16
This is how it is used: https://github.com/rusoto/rusoto/search?q=Client%3A%3Ashared

I don't know specifically what the problem is, but I can easily believe that this would cause problems.

Reply:
Since S3 does not need this change I will keep S3 as is with a note about how this is unnecessary now. I will remove the phantom from KMS because it shouldn't be necessary and even if there were a race condition in hyper it would be difficult for KMS to hit it now that it uses its own client since it very rarely makes http calls.

Reply:
[This comment](https://github.com/hyperium/hyper/issues/2112#issuecomment-630857751) indicates that avoiding the lazy_static client (connected to a global dispatcher) as the code does now fixes the issue.